cmd/ocm-backplane/cloud/common.go

```diff
@@ -12,18 +12,26 @@ import (
 	"github.com/aws/aws-sdk-go-v2/aws/arn"
 	"github.com/aws/aws-sdk-go-v2/credentials"
 	"github.com/aws/aws-sdk-go-v2/service/sts"
+	ocmsdk "github.com/openshift-online/ocm-sdk-go"
 	cmv1 "github.com/openshift-online/ocm-sdk-go/clustersmgmt/v1"
 	"github.com/openshift/backplane-cli/pkg/awsutil"
+	"github.com/openshift/backplane-cli/pkg/cli/config"
 	"github.com/openshift/backplane-cli/pkg/utils"
 )
 
 const OldFlowSupportRole = "role/RH-Technical-Support-Access"
 
-var StsClientWithProxy = awsutil.StsClientWithProxy
+var StsClientWithProxy = awsutil.StsClient
 var AssumeRoleWithJWT = awsutil.AssumeRoleWithJWT
 var NewStaticCredentialsProvider = credentials.NewStaticCredentialsProvider
 var AssumeRoleSequence = awsutil.AssumeRoleSequence
 
+// Wrapper for the configuration needed for cloud requests
+type CloudQueryConfig struct {
+	config.BackplaneConfiguration
+	OcmConnection *ocmsdk.Connection
```

Comment:
I think `OcmConnection` should be part of `config.BackplaneConfiguration` - I don't think backplane works without OCM in general (not just `ocm-backplane cloud`)

Also minor nit on the naming, if I were importing this I'd end up making a `cloud.CloudQueryConfig` or maybe I'd alias it to `backplaneCloud.CloudQueryConfig` which doesn't make too much sense/a bit too much stutter for my liking :D

Could we rename this struct to just `Config`?

Reply:
Beyond that, imo the additional struct field that's useful for the `ocm-backplane cloud` functions is either a `clusterID string` or a `cluster *cmv1.Cluster`. If a caller can assemble all the necessary backplane config file fields + OCM connection + cluster ID, then I think that's the minimum ticket needed for entry.

Reply:
> I think OcmConnection should be part of config.BackplaneConfiguration - I don't think backplane works without OCM in general (not just ocm-backplane cloud)
This is something I wanted to do, but noticed it drags too many changes in this PR. If you don't mind, I'd create a card to do this as followup (e.g. we'd have to change the whole aws interface as well then).

> Also minor nit on the naming, if I were importing this I'd end up making a cloud.CloudQueryConfig or maybe I'd alias it to backplaneCloud.CloudQueryConfig which doesn't make too much sense/a bit too much stutter for my liking :D
Could we rename this struct to just Config?

Renamed to QueryConfig. If we want to move the `QueryConfig` struct into the global backplane config in the near future, I think we should keep the clusterid out of it as we will likely have to undo that work later.